feat(controller-sdk-go): services support in controller-sdk-go

notmaxx opened the pull request.

notmaxx committed:
feat(services): add services api support to sdk.

create structures and API for calling Service REST API of controller

notmaxx committed:
ref(services): replace company includes with deis.

replace company-specific repos to deis' one

Cryptophobia reviewed (changes requested):
See conversation here for package locations: https://github.com/teamhephy/workflow-cli/pull/33#issuecomment-420754720

Cryptophobia commented on services/services.go:
```diff
@@ -0,0 +1,84 @@
+	return services, reqErr
+}
+
+// New adds a service to an app.
```
Can you provide a little more detail here for GoDocs? 

https://godoc.org/github.com/teamhephy/controller-sdk-go/apps#New

For example when creating a new app in the docs above, they talk about the newly created random name of the app. Here maybe write a bit more about procfileType and pathPattern that need to be passed to create new service object. Service is related to an application. You can't create a service without having an app first. It's obvious but I think good to be documented.

Cryptophobia commented on services/services.go:
```diff
@@ -0,0 +1,84 @@
+	return d, reqErr
+}
+
+// Delete service from app
```
Maybe write a bit more for the docs about the possible errors when trying to delete a service.

Cryptophobia replied:
Very nice, now that is good documentation! :slightly_smiling_face:

notmaxx committed:
Merge branch 'master' into services_support

notmaxx committed:
ref(services): replace deis with teamhephy

replace deis with teamhephy in import dependencies, also add more
details in comments re functionality.

notmaxx commented:
@Cryptophobia done, btw, do you have autobuild for this?

Cryptophobia reviewed (approved):
:100:

Cryptophobia commented:
> @Cryptophobia done, btw, do you have autobuild for this?

We can integrate Travis CI builds for PR testing. Nice idea.

EDIT: Adding travis.yml for ci builds in https://github.com/teamhephy/controller-sdk-go/pull/5

Cryptophobia commented:
@notmaxx , if you want to trigger travis PR build on branches now you need to merge latest from master so you get the .travis.yml file on your branch.

notmaxx committed:
Merge branch 'master' into services_support

notmaxx committed:
chore(services): format service file

use gofmt to fix formatting issues

notmaxx commented:
@Cryptophobia done

Cryptophobia reviewed (approved):
LGTM

Cryptophobia commented:
Merged!
